test/v2/core/sgETH.spec.js

```diff
@@ -53,4 +53,44 @@ describe("SgETH.sol", () => {
       .and.to.be.emit(sgEth, "RoleRevoked")
       .withArgs(ethers.constants.HashZero, deployer.address, deployer.address);
   });
+
+  it("e2e test", async () => {
+    // deploy sgeth
+    const WSGETH = await ethers.getContractFactory("WSGETH");
+    const wsgETH = await WSGETH.deploy(sgEth.address, 24 * 60 * 60);
+    await wsgETH.deployed();
+
+    const feeCalc = ethers.constants.AddressZero; // not sure if it is right
+    const numValidators = 1000;
+    const adminFee = 0;
+
+    const addresses = [
+      feeCalc,
+      sgEth.address,
+      wsgETH.address,
+      ethers.constants.AddressZero, // using dummy address
+      ethers.constants.AddressZero, // using dummy address
+    ];
+
+    // add secondary minter contract / eoa
+    const Minter = await ethers.getContractFactory("SharedDepositMinterV2");
+    const minter = await Minter.deploy(numValidators, adminFee, addresses);
+    await minter.deployed();
+
+    // revoke deployer minter rights
+    // actually don't need to do this because deployer doesn't have minter role
+    await sgEth.removeMinter(deployer.address);
+    // add secondary owner
+    // revoke deployer admin rights
+    await expect(sgEth.transferOwnership(minter.address))
+      .to.be.emit(sgEth, "RoleGranted")
+      .withArgs(ethers.constants.HashZero, minter.address, deployer.address)
+      .and.to.be.emit(sgEth, "RoleRevoked")
+      .withArgs(ethers.constants.HashZero, deployer.address, deployer.address);
+
+    // check auth invariants are preserved. i.e ex owner and outsiders cannot interact with the contract
+    await expect(sgEth.connect(deployer).transferOwnership(alice.address)).to.be.revertedWith(
```

Comment:
we check deployer cant transfer to alice, but can we also make sure alice cant transfer to anyone? deployer + 0x00?

Reply:
I think alice can transfer ownership to another. do you want me to check this?

Reply:
okay. got it